pkg/lib/synthesize/synthesis.go

```diff
@@ -0,0 +1,680 @@
+			dname = "default"
+		}
+		deploySANames[dname] = struct{}{}
+	}
+
+	// Get all RBAC w/ subjects matching Deployment ServiceAccounts by name only.
+	var permissions []v1alpha1.StrategyDeploymentPermissions
+	rolesRefAndSANameMap := make(map[rv1.RoleRef]string)
+	rbs, err := s.getRoleBindings()
+	if err != nil && err != ObjetctTypeNotFoundInBundle {
+		return nil, err
+	}
+	if rbs != nil {
+		for _, rb := range *rbs {
+			// ignore optional bindings since they are not required or provided.
+			if annotation, ok := rb.Annotations["operators.operatorframework.io.bundle.dependency"]; ok &&
+				annotation == "optional" {
+				continue
+			}
+			for _, sub := range rb.Subjects {
+				if sub.Kind != ServiceAccountKind {
+					continue
+				}
+				// Rolebinding APIGroup defaults to "" for service account subjects, therefore not monitored.
+				if _, ok := deploySANames[sub.Name]; ok {
+					rolesRefAndSANameMap[rb.RoleRef] = sub.Name
+				}
+			}
+		}
+	}
+
+	roles, err := s.getRoles()
+	if err != nil && err != ObjetctTypeNotFoundInBundle {
+		return nil, err
+	}
+	if roles != nil {
+		for _, role := range *roles {
+			gvk := role.GroupVersionKind()
+			if SAName, ok := rolesRefAndSANameMap[rv1.RoleRef{Name: role.Name, Kind: gvk.Kind, APIGroup: gvk.Group}]; ok {
+				permissions = append(permissions,
+					v1alpha1.StrategyDeploymentPermissions{ServiceAccountName: SAName, Rules: role.Rules})
+			}
+		}
+	}
+
+	var clusterPermissions []v1alpha1.StrategyDeploymentPermissions
+	clusteRolesRefAndSANameMap := make(map[rv1.RoleRef]string)
+	crbs, err := s.getClusterRoleBindings()
+	if err != nil && err != ObjetctTypeNotFoundInBundle {
+		return nil, err
+	}
+	if crbs != nil {
+		for _, crb := range *crbs {
+			// ignore optional bindings since they are not required or provided.
+			if annotation, ok := crb.Annotations["operators.operatorframework.io.bundle.dependency"]; ok &&
+				annotation == "optional" {
+				continue
+			}
+			for _, sub := range crb.Subjects {
+				if sub.Kind != ServiceAccountKind {
+					continue
+				}
+				// Rolebinding APIGroup defaults to "" for service account subjects, therefore not monitored.
+				if _, ok := deploySANames[sub.Name]; ok {
+					clusteRolesRefAndSANameMap[crb.RoleRef] = sub.Name
+				}
+			}
+		}
+	}
+
+	clusterRoles, err := s.getClusterRoles()
+	if err != nil && err != ObjetctTypeNotFoundInBundle {
+		return nil, err
+	}
+
+	if clusterRoles != nil {
+		for _, clusterRole := range *clusterRoles {
+			gvk := clusterRole.GroupVersionKind()
+			// APIGroup == gvk.Group because group is extracted from apiVersion which is group/version
+			if SAName, ok := rolesRefAndSANameMap[rv1.RoleRef{Name: clusterRole.Name, Kind: gvk.Kind, APIGroup: gvk.Group}]; ok {
+				clusterPermissions = append(clusterPermissions,
+					v1alpha1.StrategyDeploymentPermissions{ServiceAccountName: SAName, Rules: clusterRole.Rules})
+			}
+		}
+	}
+
+	return &v1alpha1.NamedInstallStrategy{
+		StrategyName: DeploymentKind,
```

Comment:
I don't think this is the right value for `StrategyName`: https://github.com/operator-framework/operator-lifecycle-manager/blob/573194a4a75f066512566e810c76f37626db3667/pkg/controller/install/resolver.go#L36